src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,28 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Checks whether or not the give entity fields are null and map to schema NULL
+     * or are not null and map to schema NOT NULL.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            $isNullable = $schema->isNullable($field);
+            $value = $entity->get($field);
+            if (!$isNullable && $value === null) {
+                return false;
+            }
+            if ($isNullable && $value !== null) {
+                return false;
```

Comment:
>  Should I add a test case for the latter?

Maybe. I don't think `exists()` would return `true` against something like the following:
```php
[
    'x' => 12345, // Any arbitrary value
    'y' => null,
]
```

> Do you know of a fixture that fits, if I can't find one I'll need to create one.

I couldn't find any suitable fixtures :cry: